iroh-gateway/src/client.rs

```diff
@@ -1,23 +1,85 @@
-// this is just a stub for future integration with iroh-gateway
+use std::{fs::File, io::Read, path::Path, time::Duration};
 
-use crate::metrics::*;
-use crate::response::ResponseFormat;
 use axum::body::Body;
 use cid::Cid;
 use metrics::{counter, gauge, histogram, increment_counter};
 use rand::{prelude::StdRng, Rng, SeedableRng};
-use std::{fs::File, io::Read, path::Path, time::Duration};
+use tracing::{error, info};
+
+use crate::metrics::*;
+use crate::response::ResponseFormat;
 
 #[derive(Debug, Clone, Copy)]
 pub struct Client {}
 
-pub const CHUNK_SIZE: usize = 1024;
+const CHUNK_SIZE: usize = 1024;
 
 impl Client {
     pub fn new() -> Self {
         Self {}
     }
 
+    #[tracing::instrument(skip(rpc_client))]
+    pub async fn get_file(
+        &self,
+        path: &str,
+        rpc_client: &iroh_rpc_client::Client,
+        start_time: std::time::Instant,
+    ) -> Result<Body, String> {
+        info!("get file {}", path);
+        let p: iroh_resolver::resolver::Path =
```

Comment:
This should be way up in `core/get_ipfs` around the definition of full content path and then pushed down the stack.

Am fine with landing this to get it going and I can pick it up in my next run over gateway.